src/utils/timeTracker/index.js

```diff
@@ -0,0 +1,188 @@
+  },
+  [CONSTANTS.MY_SEGMENTS_FETCH]: {
+    metrics: getMySegmentMetricsTracker,
+    method: 'latency'
+  },
+  [CONSTANTS.SEGMENTS_FETCH]: {
+    metrics: getSegmentChangesMetricsTracker,
+    method: 'latency'
+  },
+  [CONSTANTS.SPLITS_FETCH]: {
+    metrics: getSplitChangesMetricsTracker,
+    method: 'latency'
+  }
+};
+
+const TrackerAPI = {
+  /**
+   * "Private" method, used to attach count/countException callbacks to a promise.
+   *
+   * @param {Promise} promise - The promise we want to attach the callbacks.
+   * @param {string} task - The name of the task.
+   * @param {string} modifier - (optional) The modifier for the task, if any.
+   */
+  __attachToPromise(promise, task, modifier) {
+    return promise.then(resp => {
+      this.stop(task, modifier);
+
+      const tracker = CALLBACKS[task] && CALLBACKS[task].metrics();
+      tracker.count && tracker.count(resp.status);
+
+      return resp;
+    })
+    .catch(err => {
+      this.stop(task, modifier);
+
+      const tracker = CALLBACKS[task] && CALLBACKS[task].metrics();
+      tracker.countException && tracker.countException();
+
+      throw err;
+    });
+  },
+  /**
+   * Starts tracking the time for a given task.
+   *
+   * @param {string} task - The task we are starting.
+   * @param {Promise} promise - (optional) The promise we are tracking.
+   */
+  start(task, promise) {
+    let result;
+    // If we are registering a promise with this task, we should count the status and the exceptions as well
+    // as stopping the task when the promise resolves.
+    if (thenable(promise)) {
+      result = this.__attachToPromise(promise, task);
+    }
+
+    // Start the timer, then save the reference. We do it last to avoid counting as much extra processing
+    // as possible on the latencies.
+    timers[task] = timer();
+
+    // If no promise is present, we will return undefined as before.
+    return result;
+  },
+  /**
+   * Starts tracking the time for a given UNIQUE task. By unique task we mean tasks that fall under the same concept
+   * but each call should be tracked in a unique way. (So if you start two timers for the same task before stopping,
+   * we should track them separately)
+   *
+   * @param {string} task - The task we are starting.
+   * @return {Function} The stop function for this specific task.
+   */
+  startUnique(task, promise) {
+    const taskUniqueId = uniqueId();
+    let result;
+
+    // If we are registering a promise with this task, we should count the status and the exceptions as well
+    // as stopping the task when the promise resolves. Then return the promise
+    if (thenable(promise)) {
+      result = this.__attachToPromise(promise, task, taskUniqueId);
+    } else {
+      // If not, we return the stop function, as it will be stopped manually.
+      result = this.stop.bind(this, task, taskUniqueId);
+    }
+
+    // We start the timer, with an uniqueId attached to it's name.
+    timers[task + taskUniqueId] = timer();
```

Comment:
I guess seeing this line I'm answering myself about the multiple instance question before.